### ISSUE TITLE
EFI can not work

### ISSUE THREAD

**Issue by maomingyang1314** [Pavo-IM/OC-Gen-X#38]
Hello developer, when I tried your latest GEN X, I found that the created EFI could not boot macOS. I made EFI in full accordance with the regulations

**Comment by Pavo-IM** [comment]
Can you give me the exact settings you used I will see if I can reproduce your issue.

**Comment by maomingyang1314** [comment]
The kernel of AMD RYZEN is used. Whatevergreen.kext and sound card settings are not filled in. Enter the following code settings in the final additionalBootArgs.
npci=0x2000 keepsyms=1 debug=0x1000  alcid=1 keepsyms=1 debug=0x100 adgpmod=pikera npci=0x2000 alcid=1

**Comment by maomingyang1314** [comment]
SMBIOS has set the correct data. In the previous version, the EFI was replaced. After entering the system within five minutes, it was completely stuck and the mouse could not be moved.

**Comment by maomingyang1314** [comment]
It may be because of the AMD processor, which caused many problems. Thank you for being able to reply to my message.

**Comment by Pavo-IM** [comment]
Your bootargs should only be `npci=0x2000 keepsyms=1 debug=0x1000 alcid=1 alcid=1` not `npci=0x2000 keepsyms=1 debug=0x1000 alcid=1 keepsyms=1 debug=0x100 adgpmod=pikera npci=0x2000 alcid=1`

**Comment by maomingyang1314** [comment]
Okay, let me try again. I really appreciate your help

**Comment by Pavo-IM** [comment]
I don't have a Ryzen system to test but I did just generate a Threadripper EFI and its booting perfectly fine, it uses the same kernel patches as Ryzen except the Fix PAT patch.

**Comment by maomingyang1314** [comment]
I’m sorry, dear developer, I tried your solution. When I set Opencore as the first startup item, the computer turned on and entered the second startup item, indicating that EFI was faulty.

**Comment by maomingyang1314** [comment]
My computer’s graphics card is an AMD 5700xt graphics card. Does this graphics card require unique settings when setting it up?

**Comment by maomingyang1314** [comment]
Dear developer, I will try again tomorrow. good night

**Comment by Pavo-IM** [comment]
> My computer’s graphics card is an AMD 5700xt graphics card. Does this graphics card require unique settings when setting it up?

You will need to read the OpenCore guide on that. https://dortania.github.io/GPU-Buyers-Guide/modern-gpus/amd-gpu.html

**Comment by maomingyang1314** [comment]
Dear developers, I set up EFI this morning according to your setup method and added the graphics card option. Entering the system is normal, but within 5 minutes, the computer is completely stuck and even the mouse cannot be moved. I can only force shutdown. Replace the original EFI

**Comment by maomingyang1314** [comment]
I am also curious, my previous EFI was also created by your software. Later versions, and re-creation, will encounter this stuck problem.

**Comment by Pavo-IM** [comment]
This sounds like a HPET issue, not a EFI issue. You need to figure out why is freezing, the config is identical to the settings that the guide tells you to do manually, some AMD systems have HPET issues that you must fix yourself.

**Comment by maomingyang1314** [comment]
Very much, I will try to fix this problem

**Comment by maomingyang1314** [comment]
panic(cpu 10 caller 0xffffff7fa395df3d): CPU 15 has no HPET assigned to it
Backtrace (CPU 10), Frame : Return Address
0xffffff83bffdbc90 : 0xffffff801ff1a65d mach_kernel : _handle_debugger_trap + 0x49d
0xffffff83bffdbce0 : 0xffffff8020054a75 mach_kernel : _kdp_i386_trap + 0x155
0xffffff83bffdbd20 : 0xffffff80200465fe mach_kernel : _kernel_trap + 0x4ee
0xffffff83bffdbd70 : 0xffffff801fec0a40 mach_kernel : _return_from_trap + 0xe0
0xffffff83bffdbd90 : 0xffffff801ff19d27 mach_kernel : _DebuggerTrapWithState + 0x17
0xffffff83bffdbe90 : 0xffffff801ff1a117 mach_kernel : _panic_trap_to_debugger + 0x227
0xffffff83bffdbee0 : 0xffffff80206c1a6c mach_kernel : _panic + 0x54
0xffffff83bffdbf50 : 0xffffff7fa395df3d com.apple.driver.AppleIntelCPUPowerManagement : _pmInitThread.cold.3
0xffffff83bffdbf60 : 0xffffff7fa3948882 com.apple.driver.AppleIntelCPUPowerManagement : _pmInitThread + 0x1ad
0xffffff83bffdbfa0 : 0xffffff801fec013e mach_kernel : _call_continuation + 0x2e
      Kernel Extensions in backtrace:
         com.apple.driver.AppleIntelCPUPowerManagement(222.0)[C5BB0AFD-CB31-3FD1-BF3B-773B2919A970]@0xffffff7fa3942000->0xffffff7fa3974fff

**Comment by maomingyang1314** [comment]
The error report is like this, it should be an HPET problem, but I don’t know how to fix it

**Comment by Pavo-IM** [comment]
This is one of 2 possible issues, could be a HPET issue which requires you to make a SSDT to fix it or it could be a AppleIntelCPUPowerManagement issue which means you need to enable DummyPowerManagement under Kernel > Emulate section of the config.plist.

**Comment by maomingyang1314** [comment]
Thank you developers very much. With your help, I solved the problem. After enabling DummyPowerManagement, the previous lag problem no longer occurs

**Comment by Pavo-IM** [comment]
Good to know, closing this issue.
